src/sensor.ts

```diff
@@ -133,7 +133,7 @@ export class Sensor {
 				validationErrors.push('Other events in the batch were invalid');
 			} catch (errors) {
 				isValid = false;
-				validationErrors.push(`${event.id}: ${errors.join('\n\t')}`);
+				validationErrors.push(`${event.id}: ${(errors as Error).stack}`);
 			}
```

Comment:
We will most likely want the error message in there as well...
```suggestion
			} catch (e) {
				isValid = false;
				const e = error as Error;
				validationErrors.push(`${event.id}: ${e.message}\n${e.stack}`);
			}
```